src/gsd.ts

```diff
@@ -298,20 +298,33 @@ const getNoiseLevel = (y: DoubleArray): number => {
 
   return stddev;
 };
+
+interface Peak1DWithIndex extends Omit<Peak1D, 'index'> {
+  index: number;
+}
+
+function checkIndex(peaks: Peak1D[]): asserts peaks is Peak1DWithIndex[] {
+  for (const peak of peaks) {
+    if (peak.index === undefined) {
+      throw new Error('index property is mandatory to determine the real top');
```

Comment:
if index is mandatory, why is it optional in the types?

Reply:
In the package `nmr-processing` we extend the interface `Peak1D` to generate `NMRPeak1D` and index is not required in all of the other methods that I know.

I could make it mandatory at the level of ml-gsd because gsd returns peaks with `index` and make it optional in NMRPeak1D. are you agree?